node/src/components/consensus/consensus_protocol.rs

```diff
@@ -89,6 +89,8 @@ pub(crate) enum ProtocolOutcome<I, C: Context> {
     WeAreFaulty,
     /// We've received a unit from a doppelganger.
     DoppelgangerDetected,
+    /// We want to disconnect from a sender of invalid data.
+    Disconnect(I),
```

Comment:
In theory we could serialize the dropped vertices again, and use `InvalidIncomingMessage` instead of this one. Not sure whether it's worth the effort, though.

Reply:
We'd have to keep a message per sender when dropping them... My opinion is that it's not worth it, but I could do that if someone disagrees.

Reply:
OK; you're probably right. :+1: